.github/workflows/commit_message_checker.yml

```diff
@@ -0,0 +1,50 @@
+---
+name: 'Commit Message Check'
+# yamllint disable-line rule:truthy
+on: [push, pull_request]
+
+jobs:
+  check-commit-message:
+    name: Check Commit Message
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check Subject Begining
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^([A-Z]|\w+:)'
+          flags: 'g'
+          error: 'The subject does not start with a capital or tag.'
+          excludeDescription: 'true'
+          excludeTitle: 'true'
+          checkAllCommitMessages: 'true'
+          accessToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check Subject Line Length
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^.{1,72}(\n|$)'
+          flags: 'g'
+          error: 'The maximum subject line length of 72 characters is exceeded.'
+          excludeDescription: 'true'    # excludes the description body of a pull request
+          excludeTitle: 'true'    # excludes the title of a pull request
+          checkAllCommitMessages: 'true'    # checks all commits associated with a pull request
+          accessToken: ${{ secrets.GITHUB_TOKEN }}    # only required if checkAllCommitMessages is true
+      - name: Check Empty Line
+        uses: gsactions/commit-message-checker@v1
+        with:
+          pattern: '^.*(\n\n|$)'
+          flags: 'g'
+          error: 'No newline between title and description.'
+          excludeDescription: 'true'
```

Comment:
Have you tested this one?
I'm wondering how `excludeDescription` actually works. What is the regular expression applied to here if there are multiple lines in the commit message?

Reply:
Oh, or is this really about the github pull request comment and not related to the git commit message?

Reply:
Yes, this is about the github description. I thought that this should be left open.

I have tested this one, it's basically saying "after the first line, find either the end of the commit msg or two newlines". It will succeed with one-liners, it will succeed with multiple lines where the first line and the rest of the text are separated by an extra newline, but it will fail if the second line doesn't have an extra line between itself and the first line

Reply:
Note that it should also fail if the empty line contains spaces, which can easily be accounted for, but the extra line shouldn't contain any extra spaces anyway so I left it at that